openfisca_core/json_to_test_case.py

```diff
@@ -1,85 +1,100 @@
-                    ),
-                conv.default([]),
-                )
-
-    def get_role_parsing_dict(entity):
-        if entity.is_person:
-            return {}
+                    )
+                )(value)
+
+        if error is not None:
+            raise ValueError(u"Invalid description of {}: {}. Error: {}".format(entity_class.key, entity_json, error))
+        entity_json[key] = value
+
+    def check_variable(value, key):
+        column = tax_benefit_system.column_by_name[key]
+        if column.entity != entity_class:
+            raise ValueError(u"Variable {} is defined for entity {}. It cannot be set for entity {}.".format(key, column.entity.key, entity_class.key))
+        value, error = column.json_to_python(value)
+        if error is not None:
+            raise ValueError(u"Invalid value {} for variable {}. Error: {}".format(value, key, error))
+        entity_json[key] = value
+
+    for key, value in entity_json.iteritems():
+        if key == 'id':
+            check_id(value)
+        elif valid_roles.get(key) is not None:
+            check_role(value, key)
+        elif tax_benefit_system.column_by_name.get(key) is not None:
+            check_variable(value, key)
         else:
-            return dict(
-                build_role_parser(role)
-                for role in entity.roles
-                )
-
-    def get_entity_parsing_dict(tax_benefit_system):
-        column_by_name = tax_benefit_system.column_by_name
-        return {
-            entity.plural: conv.pipe(
-                conv.make_item_to_singleton(),
-                conv.test_isinstance(list),
-                conv.uniform_sequence(
-                    conv.test_isinstance(dict),
-                    drop_none_items = True,
-                    ),
-                conv.function(set_entities_json_id),
-                conv.uniform_sequence(
-                    conv.struct(
-                        dict(itertools.chain(
-                            dict(
-                                id = conv.pipe(
-                                    conv.test_isinstance((basestring, int)),
-                                    conv.not_none,
-                                    ),
-                                ).iteritems(),
-                            get_role_parsing_dict(entity).iteritems(),
-                            (
-                                (column.name, column.json_to_python)
-                                for column in column_by_name.itervalues()
-                                if column.entity == entity
-                                ),
-                            )),
-                        drop_none_values = True,
-                        ),
-                    drop_none_items = True,
-                    ),
-                conv.default([]),
-                )
-            for entity in tax_benefit_system.entities
-            }
+            from .taxbenefitsystems import VariableNotFound
```

Comment:
If it's really necessary to have this import here, please add a comment to explain why